src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -234,62 +238,29 @@ class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsReq
           WarningsV4(warnings.getMessage, warnings.getWarningCode.toString)))
     }
 
-    val textSentimentResult = textSentimentResultCollection.asScala.head
-    val documentSentiment = textSentimentResult.getDocumentSentiment();
-
-    val sentimentResult = if (textSentimentResult.isError) {
+    val sentimentResult = if (textSentimentResultCollection.head.isError) {
       None
     } else {
-      Some(getDocumentSentiment(documentSentiment))
+      Some(textSentimentResult.map(sentiment => getDocumentSentiment(sentiment)).toList)
     }
 
-    val error = if (textSentimentResult.isError) {
-      val error = textSentimentResult.getError
-      Some(TAErrorV4(error.getErrorCode.toString, error.getMessage, error.getTarget))
-    } else {
-      None
-    }
+    val error = if (textSentimentResultCollection.head.isError) {

```

Comment:
similar here for the error response --> if the head/first response is valid, then the error will be set as "None". What if the second/third/etc responses contain errors?